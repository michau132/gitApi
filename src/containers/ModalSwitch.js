import React, { Component, Fragment } from 'react';
import { Switch, Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import BeerListContainer from './BeerListContainer';
import Modal from '../components/Modal';
import Header from '../components/Header';
import BeerList from '../components/BeerList';
import ModalContainer from './ModalContainer';


export const WrappedBeerList = props => (
  <BeerListContainer
    {...props}
    render={beerListContainerProps => (
      <Fragment>
        <Header />
        <BeerList {...beerListContainerProps} />
      </Fragment>
    )}
  />
);


class ModalSwitch extends Component {
  static propTypes = {
    location: PropTypes.shape({
      state: PropTypes.shape({
        modal: PropTypes.bool.isRequired,
      }),
    }).isRequired,
  }

  // eslint-disable-next-line react/destructuring-assignment
  previousLocation = this.props.location;


  componentDidUpdate(nextProps) {
    const { location } = this.props;
    if (
      nextProps.history.action !== 'POP'
      && (!location.state || !location.state.modal)
    ) {
      this.previousLocation = location;
    }
  }


  render() {
    const { location } = this.props;
    const isModal = !!(
      location.state
      && location.state.modal
    );
    return (
      <Fragment>
        <Switch location={isModal ? this.previousLocation : location}>
          <Route
            exact
            path="/trainingGit/"
            component={WrappedBeerList}
          />
          <Route
            path="/trainingGit/beer/:id"
            exact
            component={WrappedBeerList}
          />
        </Switch>
        {
          isModal ? (
            <Route
              path="/trainingGit/beer/:id"
              component={props => (
                <ModalContainer
                  {...props}
                  render={store => <Modal {...store} />}
                />)}
            />
          ) : null
        }
      </Fragment>
    );
  }
}

export default ModalSwitch;
