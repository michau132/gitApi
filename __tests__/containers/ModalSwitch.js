import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ModalSwitch, { WrappedBeerList } from '../../src/containers/ModalSwitch';

const props = {
  location: {
    state: {
      modal: true,
    },
  },
};

describe('Users', () => {
  describe('component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<ModalSwitch {...props} />);
    });

    test('renders as expected', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders without crashing', () => {
      expect(wrapper).toBeDefined();
    });

    test('routes "/" to WrappedBeerList', () => {
      expect(wrapper.find('Route[exact=true][path="/"]').first().prop('component')).toBe(WrappedBeerList);
    });

    test('routes "/beer/:id" to WrappedBeerList', () => {
      expect(wrapper.find('Route[exact=true][path="/beer/:id"]').first().prop('component')).toBe(WrappedBeerList);
    });

    test('routes "/beer/:id" and should be defined', () => {
      expect(wrapper.find('Route[path="/beer/:id"]').not('[exact=true]').prop('component')).toBeDefined();
    });
  });
});
