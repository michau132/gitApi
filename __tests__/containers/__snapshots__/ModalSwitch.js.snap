// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component renders as expected 1`] = `
<Fragment>
  <Switch
    location={
      Object {
        "state": Object {
          "modal": true,
        },
      }
    }
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/beer/:id"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <Route
    component={[Function]}
    path="/beer/:id"
  />
</Fragment>
`;
